app/src/main/java/eu/laramartin/medsreminder/reports/ReportsFragment.java

```diff
@@ -0,0 +1,128 @@
+
+import static eu.laramartin.medsreminder.firebase.FirebaseUtility.getReportsReference;
+import static eu.laramartin.medsreminder.firebase.FirebaseUtility.signOut;
+
+
+public class ReportsFragment extends Fragment {
+
+    private static final String LOG_TAG = ReportsFragment.class.getCanonicalName();
+
+    @BindView(R.id.toolbar)
+    Toolbar toolbar;
+    @BindView(R.id.reports_recyclerview)
+    RecyclerView recyclerView;
+
+    private Unbinder unbinder;
+    private ChildEventListener reportsChildEventListener;
+    private ReportsAdapter reportsAdapter;
+    private DatabaseReference reportsReference;
+
+    @Nullable
+    @Override
+    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.fragment_report, container, false);
+        unbinder = ButterKnife.bind(this, view);
+        AppCompatActivity appCompatActivity = ((AppCompatActivity) getActivity());
+        appCompatActivity.setSupportActionBar(toolbar);
+        setHasOptionsMenu(true);
+
+        reportsAdapter = new ReportsAdapter();
+        recyclerView.setAdapter(reportsAdapter);
+        recyclerView.setLayoutManager(new LinearLayoutManager(getContext()));
+
+        attachDatabaseReadListener();
+        return view;
+    }
+
+    private void attachDatabaseReadListener() {
+        reportsReference = getReportsReference();
+        if (reportsChildEventListener == null) {
+            reportsChildEventListener = new ChildEventListener() {
+                @Override
+                public void onChildAdded(DataSnapshot dataSnapshot, String s) {
+                    Report snapshot = dataSnapshot.getValue(Report.class);
+                    if (snapshot != null) {
+                        reportsAdapter.add(snapshot);
+                    }
+                }
+
+                @Override
+                public void onChildChanged(DataSnapshot dataSnapshot, String s) {
+
+                }
+
+                @Override
+                public void onChildRemoved(DataSnapshot dataSnapshot) {
+
+                }
+
+                @Override
+                public void onChildMoved(DataSnapshot dataSnapshot, String s) {
+
+                }
+
+                @Override
+                public void onCancelled(DatabaseError databaseError) {
+
+                }
+            };
+            reportsReference.addChildEventListener(reportsChildEventListener);
+        }
+    }
+
+    @Override
+    public void onDestroyView() {
+        super.onDestroyView();
+        unbinder.unbind();
+        reportsReference.removeEventListener(reportsChildEventListener);
+    }
+
+
+
+    @Override
+    public void onCreateOptionsMenu(Menu menu, MenuInflater inflater) {
+        inflater.inflate(R.menu.action_bar_menu, menu);
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
```

Comment:
Move this to an abstract BaseFragment. make this fragment and the meds fragment extend the BaseFragment